python/tank/authentication/sso_saml2/utils.py

```diff
@@ -46,18 +56,107 @@ def is_sso_enabled_on_site(shotgun_module, url, http_proxy=None):
         # require authentication.
         http_proxy = _sanitize_http_proxy(http_proxy).netloc
         if http_proxy:
-            get_logger().debug("Using HTTP proxy to connect to the Shotgun server: %s" % http_proxy)
-        info = shotgun_module.Shotgun(url, session_token="dummy", connect=False, http_proxy=http_proxy).info()
-        get_logger().debug("User authentication method for %s: %s" % (url, info["user_authentication_method"]))
-        if "user_authentication_method" in info:
-            return info["user_authentication_method"] == "saml2"
+            get_logger().debug(
+                "Using HTTP proxy to connect to the Shotgun server: %s", http_proxy
+            )
+        if url not in INFOS_CACHE or ((time.time() - INFOS_CACHE[url][0]) > INFOS_CACHE_TIMEOUT):
+            get_logger().info("Infos for site '%s' not in cache or expired", url)
+            infos = shotgun_api3.Shotgun(
+                url,
+                session_token="dummy",
+                connect=False,
+                http_proxy=http_proxy
+            ).info()
+            INFOS_CACHE[url] = (time.time(), infos)
+        else:
+            get_logger().info("Infos for site '%s' found in cache", url)
+            infos = INFOS_CACHE[url][1]
     except Exception as e:
         # Silently ignore exceptions
-        get_logger().debug("Unable to connect with %s, got exception '%s' assuming SSO is not enabled" % (url, e))
+        get_logger().debug("Unable to connect with %s, got exception '%s'", url, e)
+
+    return infos
+
+
+def _get_user_authentication_method(url, http_proxy=None):
+    """
+    Get the user authentication method for site.
+
+    :param url:            Url of the site to query.
+    :param http_proxy:     HTTP proxy to use, if any.
+
+    :returns:   A string, or None, indicating which mode is used.
```

Comment:
Please mention which strings can be returned and explain why None can be returned.